### ISSUE TITLE
[BUG] AptaNet `fit()` method mutates global random state

### ISSUE THREAD

**Issue by aditi-dsi** [gc-os-ai/pyaptamer#531]
**Describe the bug**

While looking through the AptaNet estimators, I noticed that passing a `random_state` triggers `np.random.seed()` and `torch.manual_seed()` inside `fit()` method.

I understand this was likely done to ensure PyTorch backend reproducibility. However, this mutates the user's global RNG state. If someone is running this inside a larger pipeline, their script's downstream randomness gets overwritten, which breaks scikit-learn's guidelines that estimators should not mutate global state.

**To Reproduce**

This block is currently present in the `fit()` methods of both the classifier and regressor:
```python
if self.random_state is not None:
            np.random.seed(self.random_state)
            torch.manual_seed(self.random_state)
```

**Expected behavior**

I suggest we remove the global seed mutation from `fit()`. The RandomForest already properly receives `self.random_state`. For PyTorch determinism, users should set the global seed at the application level, not the library level.

Or, if we strictly want to set specifically the PyTorch state at the estimator level, we may use a context-manager approach to temporarily set the seed during `.fit()`, and then use a `try...finally` block to restore the user's original state right after `fit()` finishes.

**Versions**
<details>

0.1.0a1
</details>
